### ISSUE TITLE
IDE0031 (Null check can be simplified) - wrong fix for nullable structs

### ISSUE THREAD

**Issue by kindermannhubert** [dotnet/roslyn#18777]
**Version Used**: 15.1 (26403.7)

Having following code:
```C#
public static object M(MyStruct? x)
{
    return x == null ? null : x.Value.Value;
}

public struct MyStruct
{
    public object Value;
}
```
IDE0031 is detected and after fix it will become following:
```C#
public static object M1(MyStruct? x)
{
    return x?.Value.Value;
}
```
which does not compile (should be only "return x?.Value;").

**Comment by Pilchie** [comment]
This is a dupe of a Developer Community issue that was fixed in https://github.com/dotnet/roslyn/pull/18178.
